sde_collections/tasks.py

```diff
@@ -141,3 +143,59 @@ def resolve_title_pattern(title_pattern_id):
     TitlePattern = apps.get_model("sde_collections", "TitlePattern")
     title_pattern = TitlePattern.objects.get(id=title_pattern_id)
     title_pattern.apply()
+
+
+@celery_app.task
+def fetch_and_update_full_text(collection_id, server_type):
+    try:
+        collection = Collection.objects.get(id=collection_id)
+    except Collection.DoesNotExist:
+        raise Exception(f"Collection with ID {collection_id} does not exist.")
+
+    server_config = get_server_config(server_type)
+    token = server_config["token"]
+    url = server_config["url"]
+
+    headers = {"Content-Type": "application/json", "Authorization": f"Bearer {token}"}
+
+    payload = json.dumps(
+        {
+            "method": "engine.sql",
+            "sql": f"SELECT url1, text, title FROM sde_index WHERE collection = '/SDE/{collection.config_folder}/'",
+            "pretty": True,
+            "log": False,
+            "output": "json",
+            "resolveIndexList": "false",
+            "engines": "default",
+        }
+    )
```

Comment:
So, can you incorporate this api code into the main API function? It would also be nice if we refactor that class to be more dry -- to only instantiate to be used with a single server, like api=Api('lis_dev'). 


Here's the boilerplate code that we brainstormed.
```
class Api:
    def __init__(self, server_name: str) -> None:
        self.server_name = server_name
        self.app_name: str = server_configs[server_name]["app_name"]
        self.query_name: str = server_configs[server_name]["query_name"]
        self.base_url: str = server_configs[server_name]["base_url"]
        self.user
        self.password
        self.token

    def sql():
        if not self.token:
            raise ValueError("you must have a token to use the SQL endpoint")
    
    def get_fulltexts(self, collection):
        self.sql(f"SELECT url1, text, title FROM sde_index WHERE collection = '/SDE/{collection.config_folder}/'")
```

Make sure that you fix any other code in the codebase that calls the existing Api class.

Reply:
Changes incorporated as suggested.